modules/microshift-storage-vol-snapshot-restore.adoc

```diff
@@ -0,0 +1,79 @@
+// Module included in the following assemblies:
+//
+// microshift/creating-volume-snapshots-microshift.adoc
+
+:_content-type: PROCEDURE
+[id="restoring-vol-snapshot-microshift_{context}"]
+= Restoring a volume snapshot
+
+The following workflow demonstrates snapshot restoration. In this example, the verification steps are also given to ensure that data written to a source `PersistentVolumeClaim` (PVC) is preserved and restored on a new PVC.
+
+[IMPORTANT]
+====
+The destination PVC storage size must be equal to or greater than that of the `dataSource` PVC `VolumeSnapshot` object.
+====
+
+.Procedure
+
+. Restore a snapshot by specifying the `VolumeSnapshot` object as the `dataSource` in a `PersistentVolumeClaim` by using the following example:
++
+[source, terminal]
+----
+# oc apply -f <<EOF
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: snapshot-restore
+spec:
+  accessModes:
+  - ReadWriteOnce
+  dataSource:
+    apiGroup: snapshot.storage.k8s.io
+    kind: VolumeSnapshot
+    name: my-snap
+  resources:
+    requests:
+      storage: 1Gi
+  storageClassName: topolvm-provisioner-thin
+---
+apiVersion: v1
+kind: Pod
+metadata:
+  name: base
+spec:
+  containers:
+  - command:
+    - sh
+    - -c
+    - sleep 1d
+    image: nginxinc/nginx-unprivileged:latest
+    name: test-container
+    securityContext:
+      allowPrivilegeEscalation: false
+      capabilities:
+        drop:
+        - ALL
+    volumeMounts:
+    - mountPath: /vol
+      name: test-vol
+  securityContext:
+    runAsNonRoot: true
+    seccompProfile:
+      type: RuntimeDefault
+  volumes:
+  - name: test-vol
+    persistentVolumeClaim:
+      claimName: snapshot-restore
+EOF
+----
+
+.Verification
+
+. Wait for the pod to reach the `Ready` state:
++
+[source, terminal]
+----
+$ oc wait --for=condition=Ready pod/base
+----
+
+. Once the new pod is ready, verify that the data from your application is correct in the snapshot. This step is unique to your workflows.
```

Comment:
```suggestion
. Once the new pod is ready, verify that the data from your application is correct in the snapshot.
```

It seems a bit obvious to say that we can't tell them how to verify their application's data.